src/aggregator/aggregator/election_mgr.go

```diff
@@ -579,19 +579,28 @@ func (mgr *electionManager) checkCampaignStateLoop() {
 	defer ticker.Stop()
 
 	for {
+		select {
```

Comment:
This wait for `doneCh` is redundant as it seems to only cover the case when the manager is closed immediately after opening.

Reply:
this exact case is hit in election manager tests, where the wait between start/resign/close is minimal - if it does happen, things can deadlock, as default `campaignIsEnabled` function expects placement manager not to block.

Reply:
yeah, I just ran the tests without it and apparently I was wrong. Thanks for explaining!